src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/DefaultJsonTypeInfoResolver.cs

```diff
@@ -109,23 +115,21 @@ protected override void VerifyMutable()
             }
         }
 
-        internal static DefaultJsonTypeInfoResolver DefaultInstance
+        internal static DefaultJsonTypeInfoResolver? DefaultInstance => s_defaultInstance;
+        private static DefaultJsonTypeInfoResolver? s_defaultInstance;
```

Comment:
`internal static DefaultJsonTypeInfoResolver? DefaultInstance { get; private set; }`

Reply:
Can't use `Interlocked` APIs with properties, needs to be a field.